agents/alm.py

```diff
@@ -293,20 +293,26 @@ def _intrinsic_reward_loss(self, z, action_batch, z_next, z_next_prior, log, met
 
     def update_critic(self, z_batch, action_batch, reward_batch, z_next_batch, discount_batch, std, log, metrics):
         with torch.no_grad():
-            z_next_batch = self.model(z_batch, action_batch).sample()
             next_action_dist = self.actor(z_next_batch, std)
             next_action_batch = next_action_dist.sample()
+
             Q1_, Q2_ = self.critic_target(z_next_batch, next_action_batch)
```

Comment:
You need to use get_critic_value here as well

Reply:
Sorry sir, I realized this as well and it was fixed like 5 minutes later than the PR.